chain/client/src/stateless_validation/chunk_validator/mod.rs

```diff
@@ -690,14 +664,71 @@ impl Client {
         prev_block: &Block,
         processing_done_tracker: Option<ProcessingDoneTracker>,
     ) -> Result<(), Error> {
-        if witness.inner.chunk_header.prev_block_hash() != prev_block.hash() {
+        if witness.chunk_header.prev_block_hash() != prev_block.hash() {
             return Err(Error::Other(format!(
                 "process_chunk_state_witness_with_prev_block - prev_block doesn't match ({} != {})",
-                witness.inner.chunk_header.prev_block_hash(),
+                witness.chunk_header.prev_block_hash(),
                 prev_block.hash()
             )));
         }
 
         self.chunk_validator.start_validating_chunk(witness, &self.chain, processing_done_tracker)
     }
+
+    /// Performs state witness decoding and partial validation without requiring the previous block.
+    fn partially_validate_state_witness_in_epoch(
+        &self,
+        signed_witness: &SignedChunkStateWitness,
+    ) -> Result<ChunkStateWitness, Error> {
+        let witness = signed_witness.witness_bytes.decode()?;
+        let chunk_header = &witness.chunk_header;
+        let witness_height = chunk_header.height_created();
+        let witness_shard = chunk_header.shard_id();
+
+        if !self
+            .epoch_manager
+            .get_shard_layout(&witness.epoch_id)?
+            .shard_ids()
+            .contains(&witness_shard)
+        {
+            return Err(Error::InvalidChunkStateWitness(format!(
+                "Invalid shard_id in ChunkStateWitness: {}",
+                witness_shard
+            )));
+        }
+
+        let chunk_producer = self.epoch_manager.get_chunk_producer(
+            &witness.epoch_id,
+            witness_height,
+            witness_shard,
+        )?;
+        if witness.chunk_producer != chunk_producer {
```

Comment:
maybe my concern is benign, but would it make sense to also check if the epoch_id is the same for the epoch being processed later when validating the chunk (when the prev block is available)?

Reply:
this is a valid concern: this check is a part of `start_validating_chunk` which is executed as soon as we have previous block available

Reply:
A quick comment here saying we do later check the epoch_id would really help, but mostly this I'm guessing would be extremely hard to bluff given our calls to `epoch_manager.get_chunk_producer` etc.

Reply:
good point, added the comment in 017902df03f67f82b614dd8901f398537db7fe92